### PR TITLE
[ISSUE #2196] [Workflow] Add prometheus metrics

### LINKED ISSUE #2196: [Enhancement] Workflow Feature Implementation: Workflow execution metrics
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

Workflow Feature Implementation: Workflow execution metrics 

### Describe the solution you'd like

Workflow Feature Implementation: Workflow execution metrics 

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by horoc**
Fixes #2196 

### Motivation

#2196 

### Modifications

#2196 

### Documentation

None

### CODE REVIEW

**Comment by horoc** [comment]
hi @walterlife, could you help to review this pr?

File: eventmesh-workflow-go/internal/task/event_task.go [hunk 1]
```diff
@@ -43,6 +44,7 @@ func NewEventTask(instance *model.WorkflowTaskInstance) Task {
 }
 
 func (t *eventTask) Run() error {
+	metrics.Inc("event_task", "total")
```
[the remark this answers is not in the document]
Comment (horoc) [reply eventmesh-workflow-go/internal/task/event_task.go:47]: ok, I will update it later.

[the remark this answers is not in the document]
Comment (horoc) [reply eventmesh-workflow-go/internal/task/event_task.go]: updated

File: eventmesh-workflow-go/internal/constants/constants.go [hunk 2]
```diff
@@ -68,3 +68,23 @@ const (
...
+	MetricsSwitchTask    = "switch_task"
+	MetricsScheduler     = "scheduler"
+	MetricsEngine        = "engine"
+	MetricsTaskQueue     = "task_queue"
+
+	MetricsTotal        = "total"
+	MetricsSwitchReject = "reject"
+	MetricsSwitchPass   = "pass"
+
+	MetricsStartRequest      = "start_request"
```
Comment (walterlife) [inline eventmesh-workflow-go/internal/constants/constants.go:85]: Is it possible to initialize the metric collector first?
@horoc

File: eventmesh-workflow-go/internal/metrics/metrics.go [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+}
+
+func (p *Metrics) getCollectorByNameAndType(name string, collectorType int) prometheus.Collector {
+	switch collectorType {
+	case histogram:
+		return p.histograms[name]
+	case gauge:
+		return p.histograms[name]
+	default:
+		panic("prometheus metrics get collector error, illegal collector type")
```
Comment (walterlife) [inline eventmesh-workflow-go/internal/metrics/metrics.go:138]: try to use error return instead of panic
@horoc

Comment (horoc) [reply eventmesh-workflow-go/internal/metrics/metrics.go]: updated.

File: eventmesh-workflow-go/internal/constants/constants.go (hunk 2 above)
Comment (horoc) [reply eventmesh-workflow-go/internal/constants/constants.go:88]: updated.

**Review by walterlife — APPROVED** [review]
LGTM

**Pull request merged**